rclpy/src/rclpy/_rclpy.c

```diff
@@ -1588,6 +1588,48 @@ rclpy_wait_set_add_entity(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
+/// Destroy the waitset structure
+/**
+ * \param[in] pywait_set Capsule pointing to the waitset structure
+ * \return NULL
+ */
+static PyObject *
+rclpy_destroy_wait_set(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pywait_set;
+
+  if (!PyArg_ParseTuple(args, "O", &pywait_set)) {
+    return NULL;
+  }
+  void * p = PyCapsule_GetPointer(pywait_set, NULL);
+  if (p == Py_None) {
+    PyErr_Format(PyExc_RuntimeError, "wait set is None");
+    return NULL;
+  }
+  rcl_wait_set_t * wait_set = (rcl_wait_set_t *)p;
+  if (!wait_set) {
+    PyErr_Format(PyExc_RuntimeError, "wait set is NULL");
```

Comment:
Isnt that check redundant? This should already be checked and reported by `rcl_wait_set_fini` right ?

Reply:
Removed in 219c67410e3c75f4db97369f26204706fcb6b1ff.